client/widgets/test-execution-time/time.directive.js

```diff
@@ -0,0 +1,124 @@
+angular.module('probedock.testExecutionTimeWidget').directive('testExecutionTimeWidget', function() {
+  return {
+    restrict: 'E',
+    controller: 'TestExecutionTimeWidgetCtrl',
+    templateUrl: '/templates/widgets/test-execution-time/time.template.html',
+    scope: {
+      organization: '=',
+      test: '='
+    }
+  };
+}).controller('TestExecutionTimeWidgetCtrl', function(api, $scope, projectVersions) {
+
+  var page = 1;
+
+  $scope.pageSize = 25;
+  $scope.maxResults = 100;
+
+  $scope.params = {};
+  $scope.results = [];
+  $scope.loading = true;
+
+  $scope.chart = {
+    data: [],
+    labels: [],
+    options: {
+      pointHitDetectionRadius: 5,
+      showTooltips: false,
+      pointDot: false,
+      datasetFill: false,
+      scaleShowVerticalLines: false,
+
+      /*
+       * Fix for space issue in the Y axis labels
+       * see: https://github.com/nnnick/Chart.js/issues/729
+       * see: http://stackoverflow.com/questions/26498171/how-do-i-prevent-the-scale-labels-from-being-cut-off-in-chartjs
+       */
+      scaleLabel: function (object) {
+        return '  ' + object.value;
+      }
+    }
+  };
+
+
+  $scope.$watch('test', function(value) {
+    if (value) {
+      initFetchResults();
+    }
+  });
+
+  $scope.$watch('params', function(value) {
+    if (!_.isEmpty(value)) {
+      initFetchResults();
+    }
+  }, true);
+
+
+  $scope.fetchMore = function() {
```

Comment:
As I said before, I don't think it's a good idea to allow to fetch more results for this widget. A chart has a very limited number of results it can display before they become too close together and unreadable, unlike the test results widget were they wrap. For now, I would display a fixed number of latest results, and later add functionality to allow to change the time window (or use a zoomable time series). But a simple chart like this doesn't lend itself to displaying all results.